Build step failing with `TS2742` error

PRs with changes unrelated to `useWallet` appear to be failing at the Build step with the following message:

```
🎁 error src/hooks/useWallet.ts:8:17 - error TS2742: The inferred type of 'useWallet' cannot be named without a reference to 'ethers/node_modules/@ethersproject/providers'. This is likely not portable. A type annotation is necessary.
```

Example failure: https://github.com/Developer-DAO/web3-ui/runs/4571409517?check_suite_focus=true#step:7:15

--- a/babel.config.js
+++ b/babel.config.js
@@ -1,0 +1,11 @@
+const BABEL_ENV = process.env.BABEL_ENV;
+const cjs = BABEL_ENV === 'cjs';
+
+module.exports = {
+  presets: [
+    '@babel/preset-env',
+    '@babel/preset-typescript',
+    '@babel/preset-react',
+  ],
+  plugins: ['@babel/plugin-transform-runtime'],
+};

--- a/package.json
+++ b/package.json
@@ -15,10 +15,7 @@
   "scripts": {
     "prepare": "husky install",
     "format": "prettier --write",
-    "build:components": "yarn workspace @web3-ui/components build",
-    "build:hooks": "yarn workspace @web3-ui/hooks build",
-    "build:core": "yarn workspace @web3-ui/core build",
-    "build": "yarn build:components && yarn build:hooks && yarn build:core",
+    "build": "preconstruct build",
     "test": "jest",
     "test:watch": "yarn test --watch",
     "test:coverage": "jest --coverage --colors",

--- a/package.json
+++ b/package.json
@@ -27,7 +27,6 @@
     "publish:npm": "yarn changeset publish"
   },
   "dependencies": {
-    "@babel/cli": "^7.16.0",
     "@babel/core": "^7.16.0",
     "@babel/preset-env": "^7.16.4",
     "@babel/preset-react": "^7.16.0",

--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
   "name": "@web3-ui/root",
   "license": "MIT",
   "version": "0.0.0",
-  "private": "true",
+  "private": true,
   "engines": {
     "node": ">=16.0.0",
     "yarn": "^1.5"

--- a/package.json
+++ b/package.json
@@ -26,7 +26,7 @@
     "chromatic": "chromatic  --exit-zero-on-changes",
     "publish:npm": "yarn changeset publish"
   },
-  "devDependencies": {
+  "dependencies": {
     "@babel/cli": "^7.16.0",
     "@babel/core": "^7.16.0",
     "@babel/preset-env": "^7.16.4",

--- a/packages/components/.babelrc
+++ b/packages/components/.babelrc
@@ -1,8 +1,0 @@
-{
-  "presets": [
-    "@babel/preset-env",
-    "@babel/preset-typescript",
-    "@babel/preset-react"
-  ],
-  "plugins": ["@babel/plugin-transform-runtime"]
-}

--- a/packages/components/package.json
+++ b/packages/components/package.json
@@ -37,6 +37,7 @@
   "module": "dist/web3-ui-components.esm.js",
   "types": "dist/web3-ui-components.cjs.d.ts",
   "dependencies": {
+    "@babel/runtime": "^7.16.7",
     "@chakra-ui/icons": "^1.1.1",
     "@chakra-ui/react": "^1.7.2",
     "@emotion/react": "^11",

--- a/packages/components/package.json
+++ b/packages/components/package.json
@@ -29,7 +29,6 @@
     "dist"
   ],
   "scripts": {
-    "build": "preconstruct build",
     "format": "prettier --write \"src/**/*.{ts,tsx,json,js,jsx}\"",
     "format:check": "prettier --list-different \"src/**/*.{ts,tsx,json,js,jsx}\"",
     "prepublishOnly": "yarn build && cp ../../LICENSE ./LICENSE"

--- a/packages/components/package.json
+++ b/packages/components/package.json
@@ -37,7 +37,6 @@
   "module": "dist/web3-ui-components.esm.js",
   "types": "dist/web3-ui-components.cjs.d.ts",
   "dependencies": {
-    "@babel/runtime": "^7.16.3",
     "@chakra-ui/icons": "^1.1.1",
     "@chakra-ui/react": "^1.7.2",
     "@emotion/react": "^11",
@@ -50,7 +49,6 @@
     "react-dom": "*"
   },
   "devDependencies": {
-    "@babel/core": "^7.12.7",
     "@storybook/react": "^6.3.12",
     "@types/jest": "^26.0.15",
     "@types/node": "^16.11.9",

--- a/packages/core/.babelrc
+++ b/packages/core/.babelrc
@@ -1,8 +1,0 @@
-{
-  "presets": [
-    "@babel/preset-env",
-    "@babel/preset-typescript",
-    "@babel/preset-react"
-  ],
-  "plugins": ["@babel/plugin-transform-runtime"]
-}

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -30,7 +30,6 @@
     "dist"
   ],
   "scripts": {
-    "build": "preconstruct build",
     "format": "prettier --write \"src/**/*.{ts,tsx,json,js,jsx}\"",
     "format:check": "prettier --list-different \"src/**/*.{ts,tsx,json,js,jsx}\"",
     "prepublishOnly": "yarn build && cp ../../LICENSE ./LICENSE"

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -38,17 +38,18 @@
   "module": "dist/web3-ui-core.esm.js",
   "types": "dist/web3-ui-core.cjs.d.ts",
   "dependencies": {
+    "@babel/runtime": "^7.16.7",
     "@chakra-ui/react": "^1.7.2",
     "@emotion/react": "^11",
     "@emotion/styled": "^11",
-    "framer-motion": "^4",
     "@web3-ui/components": "^0.4.0",
-    "@web3-ui/hooks": "^0.8.1"
+    "@web3-ui/hooks": "^0.8.1",
+    "framer-motion": "^4"
   },
   "peerDependencies": {
+    "ethers": "^5.5.1",
     "react": "*",
-    "react-dom": "*",
-    "ethers": "^5.5.1"
+    "react-dom": "*"
   },
   "devDependencies": {
     "@storybook/react": "^6.3.12",

--- a/packages/core/package.json
+++ b/packages/core/package.json
@@ -38,7 +38,6 @@
   "module": "dist/web3-ui-core.esm.js",
   "types": "dist/web3-ui-core.cjs.d.ts",
   "dependencies": {
-    "@babel/runtime": "^7.16.3",
     "@chakra-ui/react": "^1.7.2",
     "@emotion/react": "^11",
     "@emotion/styled": "^11",
@@ -52,7 +51,6 @@
     "ethers": "^5.5.1"
   },
   "devDependencies": {
-    "@babel/core": "^7.12.7",
     "@storybook/react": "^6.3.12",
     "@types/classnames": "^2.2.11",
     "@types/jest": "^26.0.15",

--- a/packages/hooks/.babelrc
+++ b/packages/hooks/.babelrc
@@ -1,8 +1,0 @@
-{
-  "presets": [
-    "@babel/preset-env",
-    "@babel/preset-typescript",
-    "@babel/preset-react"
-  ],
-  "plugins": ["@babel/plugin-transform-runtime"]
-}

--- a/packages/hooks/package.json
+++ b/packages/hooks/package.json
@@ -29,7 +29,6 @@
     "dist"
   ],
   "scripts": {
-    "build": "preconstruct build",
     "format": "prettier --write \"src/**/*.{ts,tsx,json,js,jsx}\"",
     "format:check": "prettier --list-different \"src/**/*.{ts,tsx,json,js,jsx}\"",
     "test": "jest --maxWorkers=2",

--- a/packages/hooks/package.json
+++ b/packages/hooks/package.json
@@ -40,7 +40,6 @@
   "module": "dist/web3-ui-hooks.esm.js",
   "types": "dist/web3-ui-hooks.cjs.d.ts",
   "dependencies": {
-    "@babel/runtime": "^7.16.3",
     "@walletconnect/web3-provider": "^1.6.6",
     "ethers": "^5.5.1",
     "web3modal": "^1.9.4"
@@ -50,7 +49,6 @@
     "react-dom": ">=16.8.0"
   },
   "devDependencies": {
-    "@babel/core": "^7.12.7",
     "@portis/web3": "^4.0.6",
     "@storybook/addon-actions": "^6.4.0",
     "@storybook/addon-essentials": "^6.4.0",

--- a/packages/hooks/package.json
+++ b/packages/hooks/package.json
@@ -40,6 +40,7 @@
   "module": "dist/web3-ui-hooks.esm.js",
   "types": "dist/web3-ui-hooks.cjs.d.ts",
   "dependencies": {
+    "@babel/runtime": "^7.16.7",
     "@walletconnect/web3-provider": "^1.6.6",
     "ethers": "^5.5.1",
     "web3modal": "^1.9.4"

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -13,7 +13,6 @@
     "noImplicitReturns": true,
     "noUnusedLocals": true,
     "noUnusedParameters": false,
-    "outDir": "./dist",
     "pretty": true,
     "sourceMap": true,
     "strict": true,

